### DISCUSSION TITLE
Quick test on the brand new Inpainting feature

### DISCUSSION THREAD

**Discussion by arihid** [VoxelCubes/PanelCleaner#90] [General]
Hi, I just noticed that Panel Cleaner has brought a brand-new feature: Inpainting, and I eager to find it out. So, here's some result from my quick test. Everything is tested with **default, initial** settings, CUDA enabled on my RTX 3070 Laptop.

They work pretty well when the text detection can sucessfully identify texts. And I think Photoshop's Content Aware Fill can easily do things like this..

![Cuplikan layar 2024-05-02 095204 copy](https://github.com/VoxelCubes/PanelCleaner/assets/82217590/938eb38d-eaa0-4ee2-b0e2-6d15751f554e)

It can do SFX, too! Which is AWESOME! Imagine HOURS saved with this!

![Cuplikan layar 2024-05-02 095741 copy](https://github.com/VoxelCubes/PanelCleaner/assets/82217590/da94a0f0-9f7f-4c52-8b29-1ad7ffd68208)

But the text detection is a bit wonky with default settings. Well, I was always using default settings and roll with whatever it produces...

I just wonder, there's a particular scene that is a nightmare to redraw...

![Cuplikan layar 2024-05-02 100234 copy](https://github.com/VoxelCubes/PanelCleaner/assets/82217590/b4a85b5b-8f62-41cd-aff3-08945141a5bb)

And it redraws that scene SPECTACULARLY! My redrawer was struggling there!

All in all, I think it's all boils down to how to improve the text detection techniques.

**Comment by VoxelCubes** [comment]
Awesome, glad to hear about your experience! If you figure out some better settings, I can see about making them the new default.
Happy cleaning :)

**Comment by arihid** [comment]
I'm playing around with some settings, and here's what I found so far:


Text detection are detecting texts and sfxs just fine...

![gambar](https://github.com/VoxelCubes/PanelCleaner/assets/82217590/e47c0d97-b57c-4468-a619-cfd2afb74ed6)
![gambar](https://github.com/VoxelCubes/PanelCleaner/assets/82217590/da4086a0-564c-4f2f-89b8-06e352496e7e)

![gambar](https://github.com/VoxelCubes/PanelCleaner/assets/82217590/7a70ed8e-292f-4c4d-91b3-9c6ea6fa9780)

...except when they dont.

![gambar](https://github.com/VoxelCubes/PanelCleaner/assets/82217590/6129579e-ef98-4749-85c3-ee6e43b1623d)

![gambar](https://github.com/VoxelCubes/PanelCleaner/assets/82217590/6537478e-88bd-4197-a262-5b77002dd87e)

![gambar](https://github.com/VoxelCubes/PanelCleaner/assets/82217590/b4be582f-d6b0-49df-8693-6d2ab6dfefb4)
Text detection engine can't detect long horizontal bar which is common in Japanese manga.



But the preprocessor decides to discard some of that information...

![gambar](https://github.com/VoxelCubes/PanelCleaner/assets/82217590/ed8610b0-6b28-4459-adf4-d13f4a979313)
2 detected sfxs wasn't detected as textbox.


Here I tried to bring down Box Min Size to 50, Suspicious Box Min Size to 2500, and Box Overlap Threshold to 10% in Preprocessor, and leaving OCR disabled. It'll be a great idea to not filter these characters there, as some bubles may contain only that character: `. - ? !`.  I'm not saying this is a good settings, but what I'm trying here is to make Preprocessor accept those sfx. Ignoring other settings, one would expect that those ignored sfxs won't be repainted.

![gambar](https://github.com/VoxelCubes/PanelCleaner/assets/82217590/dd90c28b-c359-40d4-9072-3dd17bf7fd24)


Despite Fitment Quality step refuse to work with texts that exceeds sdev threshold, Inpainter comes into rescue.

![gambar](https://github.com/VoxelCubes/PanelCleaner/assets/82217590/eceeec5b-92e3-4479-a0fc-11fd55878cd9)
Text that Fitment Quality process refused to work with.

![gambar](https://github.com/VoxelCubes/PanelCleaner/assets/82217590/62fc86cc-a19e-44bc-a1d7-02e5c5ed5e2b)
Inpainted output. Well they are a bit wonky there. I won't blame them, I also struggling there.


If only there's a way to override Preprocessor's decision wether to include detected texts or not...

**Reply by VoxelCubes** [reply]
Thanks for all the examples. There is one mystery I can solve at least: The text detection model has 2 outputs, that black and white png as well as a list of boxes with attributes that it thinks could be text. Panel Cleaner doesn't take the black and white image and place boxes over it. Sometimes there are boxes where there is no text at all, or where there isn't even a white blob in the mask. Those are all ignored. So if a box isn't there, there likely was no box detected in the first place.

Maybe I should add something to try and add extra boxes though, hmm.

At least I should maybe visualize the raw box data, to make that more clear.

**Comment by arihid** [comment]
Hello hello, it's me again!
I just wanna say thanks. I've been out of action for a while since the manga I'm working on was in hiatus. Now that the manga is back, I also back in action.
But man, I must say I was blown away with it's capabilities now! It greatly reduced my processing time!

**Reply by VoxelCubes** [reply]
Awesome, not sure what really changed since then, but I'm happy to hear you're having a good time!
